src/main/java/org/apache/onami/persist/ResourceLocalTransactionFacadeFactory.java

```diff
@@ -36,6 +41,8 @@
    */
   private final EntityManagerProvider emProvider;
 
+  private final ThreadLocal<Outer> outerTransactionFacade = new ThreadLocal<>();
```

Comment:
if there are multiple nested transactions, this will break. I think jpa allows that.